bundles/org.openhab.binding.kvv/src/main/java/org/openhab/binding/kvv/internal/KVVStopHandler.java

```diff
@@ -126,16 +137,19 @@ public void dispose() {
     private synchronized void setDepartures(final DepartureResult departures, final int maxTrains) {
         int i = 0;
         for (; i < departures.departures.size(); i++) {
+            final DepartureResult.Departure departure = departures.departures.get(i);
+
             this.updateState(new ChannelUID(this.thing.getUID(), "train" + i + "-name"),
-                    new StringType(departures.departures.get(i).route));
+                    new StringType(departure.route.name));
             this.updateState(new ChannelUID(this.thing.getUID(), "train" + i + "-destination"),
-                    new StringType(departures.departures.get(i).destination));
-            String eta = departures.departures.get(i).time;
-            if (eta.equals("0")) {
+                    new StringType(departure.route.direction));
+            String eta = departure.eta;
+            if ("0".equals(eta)) {
                 eta += " min";
             }
             this.updateState(new ChannelUID(this.thing.getUID(), "train" + i + "-eta"), new StringType(eta));
         }
+
         for (; i < maxTrains; i++) {
```

Comment:
I have a doubt, shouldn't `i` be set to 0 ? It was incremented by the previous for loop.

Reply:
That is on purpose, because it is possible that the list of trains which is returned from the API is reduced. E.g. at night when there are less trains than at day. So we potentially need to "reset" channels which would not be updated otherwise.

Reply:
Oh yes, obviously.